frontend/packages/dev-console/src/components/topology/PodRingDataController.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { Firehose, FirehoseResource } from '@console/internal/components/utils';
+import { knativeServingResources } from '@console/knative-plugin/src/utils/create-knative-utils';
+import TransformPodRingData from './podring-utils';
+import { Resource } from './topology-types';
+
+export interface PodRingDataResources {
+  replicationControllers: Resource;
+  pods: Resource;
+  deploymentConfigs: Resource;
+  services: Resource;
+  deployments: Resource;
+  replicasets: Resource;
+  daemonSets: Resource;
+}
+
+export interface ControllerProps {
+  loaded?: boolean;
+  loadError?: any;
+  resources?: PodRingDataResources;
+  render(RenderProps): React.ReactElement;
+}
+
+export interface PodRingDataControllerProps {
+  namespace: string;
+  render(RenderProps): React.ReactElement;
+  knative: boolean;
+}
+
+const Controller: React.FC<ControllerProps> = React.memo(
+  ({ resources, render, loaded, loadError }) => {
+    return render({
+      loaded,
+      loadError,
+      data: loaded
+        ? new TransformPodRingData(resources)
+            .transformDataBy('deployments')
+            .transformDataBy('deploymentConfigs')
+            .transformDataBy('daemonSets')
+            .getPodsData()
+        : null,
+    });
+  },
+);
+
+const PodRingDataContoller: React.FC<PodRingDataControllerProps> = ({
+  namespace,
+  render,
+  knative,
+}) => {
+  let resources: FirehoseResource[] = [
+    {
+      isList: true,
+      kind: 'DeploymentConfig',
+      namespace,
+      prop: 'deploymentConfigs',
+    },
+    {
+      isList: true,
+      kind: 'Deployment',
+      namespace,
+      prop: 'deployments',
+    },
+    {
+      isList: true,
+      kind: 'Pod',
+      namespace,
+      prop: 'pods',
+    },
+    {
+      isList: true,
+      kind: 'ReplicaSet',
+      namespace,
+      prop: 'replicasets',
+    },
+    {
+      isList: true,
+      kind: 'ReplicationController',
+      namespace,
+      prop: 'replicationControllers',
+    },
+    {
+      isList: true,
+      kind: 'DaemonSet',
+      namespace,
+      prop: 'daemonSets',
+    },
+    {
+      isList: true,
+      kind: 'Service',
+      namespace,
+      prop: 'services',
+    },
+  ];
+
+  if (knative) {
+    const knativeResource = knativeServingResources(namespace);
+    resources = [...resources, ...knativeResource];
+  }
+
+  return (
+    <Firehose resources={resources} forceUpdate>
```

Comment:
do we need to fetch resources via `Firehose` as in overview we already have all resources data even for knative